src/main/java/io/supertokens/webserver/api/dashboard/VerifyDashboardUserSessionAPI.java

```diff
@@ -57,15 +58,23 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws I
 
         sessionId = Utils.normalizeAndValidateStringParam(sessionId, "sessionId");
         try {
-            if (Dashboard.isValidUserSession(
-                    super.getAppIdentifierWithStorageFromRequestAndEnforcePublicTenant(req), main, sessionId)) {
-                JsonObject response = new JsonObject();
-                response.addProperty("status", "OK");
-                super.sendJsonResponse(200, response, resp);
+            JsonObject invalidSessionResp = new JsonObject();
+            invalidSessionResp.addProperty("status", "INVALID_SESSION_ERROR");
+            AppIdentifierWithStorage identifierWithStorage = super.getAppIdentifierWithStorageFromRequestAndEnforcePublicTenant(req);
+
+            if (Dashboard.isValidUserSession(identifierWithStorage, main, sessionId)) {
+                String email = Dashboard.getEmailFromSessionId(identifierWithStorage, main, sessionId);
+
+                if (email == null) {
+                    super.sendJsonResponse(200, invalidSessionResp, resp);
+                } else {
+                    JsonObject response = new JsonObject();
```

Comment:
add email only if cdi in request object is greater than equal to <same as multi tenancy version>